File: Pyramids/hollowPyramidWithCross/hollowPyramidWithCross.py
numRows = 7
temp = 1
patternSymbol = '*'

for rowVal in range(1, numRows + 1):
    for colVal in range(1, (numRows - rowVal) + 1):
        print("  ", end="")

    for colVal in range(1, temp + 1):
        if colVal == 1 or rowVal == numRows or colVal == temp:
            print(patternSymbol, "", end="") # heer, I am only using this "" because, sep parameter is already giving us one space as sep=" "
        elif colVal == rowVal or rowVal == (numRows//2) + 1:
            print(patternSymbol, "", end="")
        else:
            print("  ", end="")
    
    temp += 2;  
    print()   # for new line

File: Star Patterns/Pyramids/hollowPyramidWithCross/hollowPyramidWithCross.cpp
#include <iostream>
using namespace std;

int main() {
    int numRows = 7;
    int rowVal;
    int colVal;
    int temp = 1;
    char patternSymbol = '*';

    for(rowVal = 1; rowVal <= numRows; rowVal++) {
        for(colVal = 1 ;colVal <= numRows - rowVal; colVal++) {
            cout<<"  ";
        } 
        for(colVal = 1; colVal <= temp; colVal++) {
            if(colVal == 1 || rowVal == numRows || colVal == temp) {
                cout<<patternSymbol<<" ";
            }else if (colVal == rowVal || rowVal == (numRows/2) + 1) {
                cout<<patternSymbol<<" ";
            } else {
                cout<<"  ";
            }
        }
        temp += 2;
        cout<<"\n";
    }
    return 0;
}
